rosbag2_transport/test/rosbag2_transport/test_record.cpp

```diff
@@ -141,3 +141,37 @@ TEST_F(RecordIntegrationTestFixture, records_sensor_data)
   EXPECT_FALSE(recorded_messages.empty());
 }
 #endif  // ROSBAG2_ENABLE_ADAPTIVE_QOS_SUBSCRIPTION
+
+TEST_F(RecordIntegrationTestFixture, topic_qos_overrides)
+{
+  auto strict_msg = std::make_shared<test_msgs::msg::Strings>();
+  strict_msg->string_value = "strict";
+  std::string strict_topic = "/strict_topic";
+
+  rosbag2_transport::RecordOptions record_options =
+  {false, false, {strict_topic}, "rmw_format", 100ms};
```

Comment:
Consider using aggregated initialization instead: https://en.cppreference.com/w/cpp/language/aggregate_initialization
I.e. `{ .field1 = ..., .field2 = ...}`.

For those `struct` containing options which are mainly primitive types, you're really at risk of failing to the usual blunder: someone will add a new field somewhere, and all assignments are shifted by one, and the last attribute is uninitialized. Stating the fields explicitly at least guarantee you that each value won't end up being shifted.

Reply:
So it doesn't look like we can use this in C++14?

```
 error: ISO C++ does not allow C99 designated initializers [-Werror=pedantic]
```

Reply:
Correct, this is a C99 syntax and is not supported in C++ until C++20